tests/pexpects/autosuggest.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+from pexpect_helper import SpawnedProc
+
+sp = SpawnedProc()
+send, sendline, sleep, expect_prompt = (
+    sp.send,
+    sp.sendline,
+    sp.sleep,
+    sp.expect_prompt,
+)
+
+
+def use_suggestion(*, delay=0.1):
+    sleep(delay)
+    send("\033[C")
+    sendline("")
+
+
+def run(cmd: str):
+    sendline(cmd)
+    expect_prompt()
+
+
+expect_prompt()
+
+run("echo Hello")
+
+# basic samecase history suggestion
+send("echo He")
+use_suggestion()
+expect_prompt(">echo Hello\r\nHello")
+
+# case-correcting history suggestion
+send("echo he")
+use_suggestion()
+expect_prompt(">echo Hello\r\nHello")
+
+# prefer samecase history suggestions, even if older
+run("echo hello")
+send("echo He")
+use_suggestion()
+expect_prompt(">echo Hello\r\nHello")
+
+# case-correcting command suggestion
+send("Tru")
+use_suggestion(delay=2.0)
+expect_prompt(">true \r\n")
+
+# the motivating example: prefer icase history suggestions over icase completion suggestions
+run("mkdir -p Projects/myproject Projects/wrongproject")
```

Comment:
The existence of "Projects/wrongproject" suggests that this would be autosuggested
before the patch. But it's actually not about that,
only about making sure that "cd Projects/" file-based autosuggestion is not suggested
over history-based smartcase (or just icase?) autosuggestion,
so do we need this for anything? It's also listed last in lexicographic order,
so it wouldn't be autosuggested on `cd Projects/` either -- I'd say either find a use for it or remove it.